sdk/test-utils/recorder/README.md

```diff
@@ -1,9 +1,450 @@
+update their recordings. This way, you can focus on fixing a specific set of
+test with `.only`, then remove all the `.only` calls and trust that the playback will
+keep confirming that the unaffected tests are fine and green.
+
+### Securing sensitive data
+
+Live tests need to do sensitive operations, like authenticating with your Azure
+credentials.  To protect this information from the recordings, the
+test-utils-recorder provides the following functions:
+
+- `setReplaceableVariables`, which will allow you to hide sensitive content
+  from the environment variables (more on that later).
+- `setReplacements`, which will allow you to hide sensitive content by doing
+  pattern matching in the output files.
+- `skipQueryParams`, since query parameters may contain sensitive information,
+  the array provided to method will signal what query parameters to remove from
+  the recordings.
+
+Let's see some examples.
+
+#### setReplaceableVariables
+
+This method will allow you to specify what environment variables have sensitive information.
+Give to this function a plain object composed of the names of the environment
+variables that can't be stored in the recordings, with the values you want to
+store instead. To solve this issue in our previous example test, you may do the following:
+
+```typescript
+// You'll be most certainly importing more than this method.
+import { setReplaceableVariables } from "@azure/test-utils-recorder";
+
+// Do this inside of the beforeEach, before you start recording.
+setReplaceableVariables({
+  AZURE_CLIENT_ID: "azure_client_id",
+  AZURE_CLIENT_SECRET: "azure_client_secret",
+  AZURE_TENANT_ID: "azure_tenant_id",
+  KEYVAULT_NAME: "keyvault_name"
+});
+```
+
+If you record the tests with that change, you should see the values of those
+environment variables replaced with their safe counterpart.
+
+#### setReplacements
+
+A more generic way to clean up your recordings is by providing you with the
+final string of the recording, to which you'll be able to make arbitrary
+changes. The `setReplacements` method expects to receive an array of functions, all of which will receive
+the recorded string and return a string.
+
+Let's say your project can't include the word `mango`. You will be able to get rid of it with the following code inside your `beforeEach`:
+
+```typescript
+setReplacements([
+  (recording: string): string =>
+    recording.replace(/mango/g, "bananas"),
+]);
+```
+
+#### skipQueryParams
+
+Some HTTP requests migth have parameters with sensitive information. To get
+them out of your recordings, you can call to `skipQueryParams` with a key-value
+plain object where you specify the name of the query parameter you want to replace,
+and the value you want to end up storing, as follows:
+
+```typescript
+skipQueryParams({
+  myQueryParameter: "theValueIWantStoredInMyRecordings",
+  anotherQueryParameter: "theOtherValueIWantStoredInMyRecordings",
+});
+```
```

Comment:
Nope. 

An example for the current `skipQueryParams` method -

[Query params for some random request]
`?sv=2018-11-09&sr=c&sig=<sig>&sktid=<sktid>&skv=2018-11-09&se=2019-08-07T07%3A00%3A00Z&sp=rwdl`.

Suppose we want to skip "sr", "sig", "sp" params from the above.  
`skipQueryParams(["sr", "sig", "sp"])` can be called to remove those query parameters from the recorded request.

Reply:
Thank you!